internal/notifier/google_pubsub_topic_test.go

```diff
@@ -0,0 +1,139 @@
+	expectedPostErr   error
+	expectedCloseErr  error
+}
+
+func (tt *googlePubSubTestCase) Publish(ctx context.Context, topicID string, msg []byte) (serverID string, err error) {
+	if topicID != tt.expectedTopicID {
+		return "", fmt.Errorf("topicID does not match expected. want %v, got %v", tt.expectedTopicID, topicID)
+	}
+	if strMsg := string(msg); strMsg != tt.expectedMessage {
+		return "", fmt.Errorf("msg does not match expected. want %v, got %v", tt.expectedMessage, strMsg)
+	}
+	// serverID is only used in a debug log for now, there's no way to assert it
+	return "", tt.publishErr
+}
+
+func (tt *googlePubSubTestCase) Close() error {
+	return tt.expectedCloseErr
+}
+
+func TestGooglePubSubTopic(t *testing.T) {
+	for name, tt := range map[string]*googlePubSubTestCase{
+		"invalid topic names return error": {
+			topicName:        "invalid-topic-name",
+			expectedParseErr: errors.New("topic name doesn't match 'projects/<projectID>/topics/<topicID>': 'invalid-topic-name'"),
+		},
+		"project and topic ids are parsed from the topic name": {
+			topicName:         "projects/parsedProjectID/topics/parsedTopicID",
+			expectedProjectID: "parsedProjectID",
+			expectedTopicID:   "parsedTopicID",
+			expectedMessage:   `{"involvedObject":{},"severity":"","timestamp":null,"message":"","reason":"","reportingController":""}`,
+		},
+		"commit status updates are dropped": {
+			topicName:         "projects/projectID/topics/topicID",
+			expectedProjectID: "projectID",
+			expectedTopicID:   "topicID",
+			event: eventv1.Event{
+				Metadata: map[string]string{"commit_status": "update"},
+			},
+		},
+		"events are properly marshaled": {
+			topicName:         "projects/projectID/topics/topicID",
+			expectedProjectID: "projectID",
+			expectedTopicID:   "topicID",
+			event: eventv1.Event{
+				Metadata: map[string]string{"foo": "bar"},
+			},
+			expectedMessage: `{"involvedObject":{},"severity":"","timestamp":null,"message":"","reason":"","metadata":{"foo":"bar"},"reportingController":""}`,
+		},
+		"publish error is relayed": {
+			topicName:         "projects/projectID/topics/topicID",
+			expectedProjectID: "projectID",
+			expectedTopicID:   "topicID",
+			expectedMessage:   `{"involvedObject":{},"severity":"","timestamp":null,"message":"","reason":"","reportingController":""}`,
+			publishErr:        errors.New("publish error"),
+			expectedPostErr:   fmt.Errorf("error publishing event to topic projects/projectID/topics/topicID: %w", errors.New("publish error")),
+		},
+		"close error is relayed": {
+			topicName:         "projects/projectID/topics/topicID",
+			expectedProjectID: "projectID",
+			expectedTopicID:   "topicID",
+			expectedMessage:   `{"involvedObject":{},"severity":"","timestamp":null,"message":"","reason":"","reportingController":""}`,
+			expectedCloseErr:  errors.New("close error"),
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			projectID, topicID, err := ParseGooglePubSubTopic(tt.topicName)
```

Comment:
I don't think it's wise to test multiple behaviours in a single test case. Please split out the tests for `ParseGooglePubSubTopic` into a separate case. This makes it much easier to track down test failures and speeds up testing of separate components.

Reply:
Fixed, thanks!